experimental/Schemes/Tjurina.jl

```diff
@@ -0,0 +1,770 @@
+
+Return the `k`-th local Tjurina number of `(V(f),p)` at `p`. 
+By default computes the local Tjurina number (`k=0`) at `p`.
+Higher Tjurina numbers are of interest in positive characteristic. 
+Return -1 if infinite.
+# Examples
+```jldoctest
+julia> R,(x,y) = QQ["x", "y"];
+
+julia> L,_  = localization(R, complement_of_point_ideal(R, [0, 0]));
+
+julia> f = L(x*y*(x-1));
+
+julia> tjurina_number(f)
+1
+```
+"""
+function tjurina_number(f::MPolyLocRingElem{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, k::Integer = 0)
+  X = HypersurfaceGerm(quo(parent(f), ideal(parent(f), f))[1])
+  return tjurina_number(X, k)
+end
+
+################################################################################
+
+#####                                 Order                                #####
+
+################################################################################
+
+
+function _order(f::MPolyRingElem)                                                                                 #TODO:  export???
+  f != 0 || return -1
+  x = gens(parent(f))
+  n = length(x)
+  degrees = [sum([degree(m, j) for j in 1:n]) for m in monomials(f)]
+  return minimum(degrees)
+end
+
+@doc raw"""
+    order_as_series(f::MPolyLocRingElem{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal})
+
+Return the order of the series expansion of an element of a local ring at the localized point.
+Return -1 if infinite.
+# Examples
+```jldoctest
+julia> R,(x,y) = QQ["x", "y"];
+
+julia> L0,_  = localization(R, complement_of_point_ideal(R, [0, 0]));
+
+julia> L1,_  = localization(R, complement_of_point_ideal(R, [1, 0]));
+
+julia> f = (x-1)^3
+x^3 - 3*x^2 + 3*x - 1
+
+julia> order_as_series(L0(f))
+0
+
+julia> order_as_series(L1(f))
+3
+```
+"""
+function order_as_series(f::MPolyLocRingElem{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal})                       
+  shift, _ = base_ring_shifts(parent(f))
+  return _order(shift(numerator(f)))           
+end
+
+
+
+
+###############################################################################
+
+#####                         Finite determinacy                          #####
+
+###############################################################################
+@doc raw"""
+    is_finitelydetermined(f::MPolyLocRingElem{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, equivalence::Symbol = :contact)
+
+Return if 'f' is finitely determined w.r.t. ':right' or ':contact' equivalence.
+By default computes w.r.t. contact equivalence.
+# Examples
+```jldoctest
+julia> R,(x,y) = QQ["x", "y"];
+
+julia> L,_  = localization(R, complement_of_point_ideal(R, [0, 0]));
+
+julia> is_finitelydetermined(L(x^3-y^2))
+true
+
+julia> is_finitelydetermined(L(x^3-y^2), :right)
+true
+```
+"""
+function is_finitelydetermined(f::MPolyLocRingElem{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, equivalence::Symbol = :contact)
```

Comment:
Should be like this following our conventions
```suggestion
function is_finitely_determined(f::MPolyLocRingElem{<:Field, <:Any, <:Any, <:Any, <:MPolyComplementOfKPointIdeal}, equivalence::Symbol = :contact)
```